src/loginServices.js

```diff
@@ -360,38 +363,41 @@ function createOkapiSession(okapiUrl, store, tenant, token, data) {
  * if successful, dispatch the result to create a session
  * if not, clear the session and token.
  *
- * @param {} okapiUrl
- * @param {*} store
- * @param {*} tenant
- * @param {*} session
+ * @param {string} okapiUrl
+ * @param {redux store} store
+ * @param {string} tenant
+ * @param {object} session
  *
- * @return Promise
+ * @returns {Promise}
  */
 function validateUser(okapiUrl, store, tenant, session) {
   return fetch(`${okapiUrl}/bl-users/_self`, { headers: getHeaders(tenant, session.token) }).then((resp) => {
-    if (resp.status >= 400) {
+    if (resp.ok) {
+      const { token, user, perms } = session;
+      return resp.json().then((data) => {
+        store.dispatch(setLoginData(data));
+        store.dispatch(setSessionData({ token, user, perms }));
+        return loadResources(okapiUrl, store, tenant, user.id);
```

Comment:
@zburke not sure why but it feels wrong to me to handle loading resources inside `createOkapiSession`. I would expect `createOkapiSession` to just create the session. I wonder if we could instead load them inside `processOkapiSession`.